cpp/cmake_modules/FindArrowPython.cmake

```diff
@@ -46,8 +46,9 @@ endif()
 find_package(Arrow ${find_package_arguments})
 
 if(ARROW_FOUND)
+  message(STATUS "CPYARROW_HOME: ${CPYARROW_HOME}")
   arrow_find_package(ARROW_PYTHON
-                     "${ARROW_HOME}"
+                     "${CPYARROW_HOME}"
```

Comment:
This smells a bit hackish. Feels a bit like we are building two totally different projects. 

On long term we should probably have `libarrow_python.so` just be one of the shared objects constituting pyarrow. `libarrow_python.so` should probably be possible to integrate with the rest of the built files using `add_subdirectory` or something equivalent.

Reply:
> On long term we should probably have libarrow_python.so just be one of the shared objects constituting pyarrow.

That's actually what is already happening, I think. The libarrow_python.so gets copied into the pyarrow directory (either the repo itself for an inplace build, or the build directory, before that one gets copied to site-packages for a normal install). 
But so for building the cython extensions, we need to point to where libarrow_python.so can be found (which is now a different location as where libarrow.so can be found). And doing that with a argument to cmake seems a good way to do that? (alternatively we might need to edit the `arrow_find_package` to work with this new situation?)